python/src/deltachat/account.py

```diff
@@ -222,6 +223,25 @@ def get_message_by_id(self, msg_id):
         """ return Message instance. """
         return Message.from_db(self._dc_context, msg_id)
 
+    def get_mime_headers(self, msg_id):
+        """ return mime-header object for an incoming message.
+
+        This only returns a non-None object if ``save_mime_headers``
+        config option was set and ``msg_id`` refers to an incoming
+        message.
+
+        :param msg_id: integer message id
+        :returns: email-mime message object.
+        """
+        import email.parser
+        mime_headers = lib.dc_get_mime_headers(self._dc_context, msg_id)
+        if mime_headers:
+            s = ffi.string(mime_headers)
+            if isinstance(s, bytes):
+                s = s.decode("ascii")
```

Comment:
not sure if this is fully correct but it works in my setup. i think the char-string returned from dc_get_mime_headers is an ascii-string -- unicode chars in mime-header values have a special encoding (dealt with by the email parsing message object which this function returns)